ddtrace/tracer/tracer.go

```diff
@@ -171,9 +190,19 @@ func (t *tracer) worker() {
 			t.pushPayload(trace)
 
 		case <-ticker.C:
+			if t.statsd != nil {
+				tags := t.statsTags()
+				tags = append(tags, "reason:timeout")
+				t.statsd.Incr("datadog.tracer.flush.count", tags, 1)
+			}
 			t.flushPayload()
 
 		case confirm := <-t.flushChan:
+			if t.statsd != nil {
+				tags := t.statsTags()
+				tags = append(tags, "reason:payload-full")
+				t.statsd.Incr("datadog.tracer.flush.count", tags, 1)
+			}
```

Comment:
I've added these two sections here to differentiate between a flush due to a timeout and a flush due to a full payload, but I'm not that happy with them here. I wish we could put this in `flushPayload` itself, but `flushPayload` doesn't know why it was called.

We could pass it a value to tell it why, but adding function arguments only to be used in statistics seems undesirable. 

One additional problem is that this code assumes a message on flushChan is due to a full payload. Currently that's the only reason we send a message on that channel, but it might not always be the case. I considered calling it `"reason:triggered"`, but that seems like it's being unnecessarily vague just in case things change someday. `"reason:triggered"` seems less useful to someone using the tracer who may not know the internals than `"reason:payload-full"`

Reply:
I think it's fine to have them here. Once you apply the other comments (creating the interface and setting up the tags from the git go) this will become one line:
```
t.statsd.Incr("datadog.trace.flush.count", []string{"reason:size"}, 1)
```
That's it. Tags are added automatically and if statsd is disabled the no-op one will do nothing.

Lastly, the reason here is not that the payload is getting full, it's that it's getting larger than half of the maximum payload size (10/2 = 5MB) and should be flushed to avoid having it rejected by the agent.